source/Calamari/Kubernetes/ResourceStatus/Resources/Pod.cs

```diff
@@ -1,36 +1,121 @@
+using System.Linq;
 using Newtonsoft.Json.Linq;
 
 namespace Calamari.Kubernetes.ResourceStatus.Resources
 {
     public class Pod : Resource
     {
-        public string Phase { get; }
+        public string Status { get; }
         public override ResourceStatus ResourceStatus { get; }
     
         public Pod(JObject json) : base(json)
         {
-            Phase = Field("$.status.phase");
-            
-            // TODO implement this
-            switch (Phase)
+            var phase = Field("$.status.phase");
+            var initContainerStatuses = data
+                .SelectToken("$.status.initContainerStatuses")
+                ?.ToObject<ContainerStatus[]>() ?? new ContainerStatus[] { };
+            var containerStatuses = data
+                .SelectToken("$.status.containerStatuses")
+                ?.ToObject<ContainerStatus[]>() ?? new ContainerStatus[] { };
+
+            (Status, ResourceStatus) = GetStatus(phase, initContainerStatuses, containerStatuses);
+        }
+    
+        public override bool HasUpdate(Resource lastStatus)
+        {
+            var last = CastOrThrow<Pod>(lastStatus);
+            return last.ResourceStatus != ResourceStatus || last.Status != Status;
+        }
+
+        private (string, ResourceStatus) GetStatus(
+            string phase, 
+            ContainerStatus[] initContainerStatuses,
+            ContainerStatus[] containerStatuses)
+        {
+            switch (phase)
             {
-                case"Succeeded": 
-                case "Running":
-                    ResourceStatus = ResourceStatus.Successful;
-                    break;
                 case "Pending":
-                    ResourceStatus = ResourceStatus.InProgress;
-                    break;
+                    if (!initContainerStatuses.Any() && !containerStatuses.Any())
+                    {
+                        return ("Pending", ResourceStatus.InProgress);
+                    }
+                    return initContainerStatuses.All(HasCompleted) 
+                        ? GetStatus(containerStatuses) 
+                        : GetInitializingStatus(initContainerStatuses);
+                case "Failed":
+                    return (GetReason(containerStatuses.FirstOrDefault()), ResourceStatus.Failed);
+                case "Succeeded":
+                    return (GetReason(containerStatuses.FirstOrDefault()), ResourceStatus.Successful);
                 default:
-                    ResourceStatus = ResourceStatus.Failed;
-                    break;
+                    return ("Running", ResourceStatus.Successful);
```

Comment:
Should this return `ResourceStatus.InProgress`

Reply:
This is perhaps something worth discussion. I think Successful makes the most sense for most scenarios. In a typical deployment, a container runs a long-running process and will never terminate unless error occurs (thinking a web server). In this scenario, the final state that the user desires is that all Pods are in a "Running" state. In other words, if all Pods are "Running", for the deployment that's a success. So `Successful` reflects the status better than `InProgress`.

Reply:
makes sense